server/datastore/mysql/migrations/tables/20211102135149_AddHostSeenTimesTable.go

```diff
@@ -0,0 +1,39 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20211102135149, Down_20211102135149)
+}
+
+func Up_20211102135149(tx *sql.Tx) error {
+	hostSeenTimesTable := `
+		CREATE TABLE IF NOT EXISTS host_seen_times (
+			host_id int(10) UNSIGNED NOT NULL,
+			seen_time timestamp NULL DEFAULT NULL,
+			PRIMARY KEY (host_id),
+			INDEX idx_host_seen_times_seen_time (seen_time)
```

Comment:
+1, with the ON DELETE CASCADE.

Reply:
It is deliberate. Otherwise, inserting in bulk into `host_seen_times` will lock the `hosts` table because of the foreign keys. And locking is the main reason we are moving this out.